core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -83,6 +85,7 @@ protected SerializableTable(Table table) {
     this.encryption = table.encryption();
     this.locationProvider = table.locationProvider();
     this.refs = SerializableMap.copyOf(table.refs());
+    this.metricsReporter = table.metricsReporter();
```

Comment:
Can we expose this through `BaseTable` instead of the `Table` interface? I'm not sure that we want this in the `Table` interface and it would be best to avoid it.